### DISCUSSION TITLE
HA Thermostat precision and F to C conversion

### DISCUSSION THREAD

**Discussion by JulianGroeli** [gysmo38/mitsubishi2MQTT#209] [Q&A] [answered]
What level of precision does are the temperature numbers handled at within Mitsubishi2MQTT?

I'm using external temp sensors that report temp in Fahrenheit with 0.1 precision: eg. 71.3 deg
The home assistant thermostat just reports temps to the nearest degree F, but sometimes it seems like it's rounding the wrong way.

I tried setting the 'temp step' property to 0.1. Home assistant now allows setting the temp in 0.1 deg increments, but it still reports the current temp in whole degrees, and after a while the set temp resets back to the nearest degree.

Maybe there are rounding errors introduced when going from F to C and back?

Is there any way improve precision?

**Comment by Sprinterfreak** [comment, marked as answer]
I'm in the Celsius part of the planet. I feed remote temperatures from another µC (Espeasy) via mqtt with .01 accuracy.
mitsubishi2MQTT or the heatpump itself only reports back an accuracy up to .5 degrees.

This is how Celsius to Fahrenheit conversion is implemented:
https://github.com/gysmo38/mitsubishi2MQTT/blob/16a5821fd441e6f0c97a07785c1cd1a3ca0a07b3/src/mitsubishi2mqtt/mitsubishi2mqtt.ino#L1692

**Reply by JulianGroeli** [reply]
Yeah unfortunately we are still struggling along with multiple systems of measurement over here. :)

Thanks for pointing me to that code. The conversion math seems correct, so it's likely that there's a rounding error going on.

Maybe it's related to issue #200: set temperature off by one degree?

Unfortunately I'm pretty new to this so not sure how I can help.
